git-config/hooks/pre-commit

```diff
@@ -18,4 +18,13 @@ done
 pyflakes3 /usr/local/ub_tools/cronjobs
 
 
+# Format C++ source files:
+readonly changed_files=$(git diff --cached --name-only --diff-filter=ACM)
+readonly GIT_ROOT=$(git rev-parse --show-toplevel)
+for changed_file in changed_files; do
+    # "-style=file" has the effect of searching up the directory tree until a file named ".clang-format" is found and then using it.
+    clang-format-12 -style=file -i "$GIT_ROOT/$changed_file"
```

Comment:
maybe we should leave this inactive for now so we have a chance to finally test your changes before enabling it in the hook.

Reply:
Done!